doc/contributing/coding-conventions.chapter.md

```diff
@@ -2,12 +2,59 @@
 
 ## Syntax {#sec-syntax}
 
-- Use 2 spaces of indentation per indentation level in Nix expressions, 4 spaces in shell scripts.
+The key words _must_, _must not_, _required_, _shall_, _shall not_, _should_, _should not_, _recommended_, _may_, and _optional_ in this section are to be interpreted as described in [RFC 2119](https://tools.ietf.org/html/rfc2119). Only _emphasized_ words are to be interpreted in this way.
+
+The nixpkgs repository is formatted with [nixpkgs-fmt](https://github.com/nix-community/nixpkgs-fmt/).
+
+A non-exhaustive list of formatting rules follows. When in doubt, the tool is authoritative.
+
+- Use 2 spaces of indentation in Nix expressions, 4 spaces in shell scripts.
 
-- Do not use tab characters, i.e. configure your editor to use soft tabs. For instance, use `(setq-default indent-tabs-mode nil)` in Emacs. Everybody has different tab settings so it’s asking for trouble.
+- Do not use tab characters, i.e. configure your editor to use soft tabs. Everybody has different tab settings so it’s asking for trouble.
```

Comment:
```suggestion
- Do not use tab characters, i.e. configure your editor to use soft tabs or replace tabs with two spaces. Please refer to your editors manual on exact details. Everybody has different tab settings so it’s asking for trouble.
```

Reply:
What do you mean here with "replace tabs with two spaces"? When should a user do that, as opposed to configuring their editor?